File: cpp/program85.cpp
// program77.c

#include<iostream>
using namespace std;

class ArrayX
{
    public:
        int *Arr;
        int iSize;

        ArrayX(int);

        void Accept();
        void Display();
        void DisplayEven();
};

ArrayX::ArrayX(int X)
{
    iSize = X;
    Arr = new int[iSize];
}

void ArrayX::Accept()
{
    cout<<"Enter elements of the array \n";
    for(int icnt = 0; icnt < iSize; icnt++)
    {
        cin>>Arr[icnt];
    }
}

void ArrayX::Display()
{
    cout<<"Elements of the array are \n";
    for(int iCnt = 0; iCnt < iSize; iCnt++)
    {
        cout<<Arr[iCnt]<<"\t";
    }
    cout<<"\n";
}

void ArrayX::DisplayEven()
{
    cout<<"Even elements of the array are :\n";
    for(int iCnt = 0; iCnt < iSize; iCnt++)
    {
        if((Arr[iCnt] % 2) == 0)
        {
            cout<<Arr[iCnt]<<"\t";
        }
    }
    cout<<"\n";
}

int main()
{
    int iSize = 0;

    cout<<"Enter number of elements \n";
    cin>>iSize;

    ArrayX obj(iSize);
    obj.Accept();
    obj.Display();
    obj.DisplayEven();

    return 0;
}

File: java/program85.java
import java.util.*;

class ArrayX
{
    public int Arr[];

    ArrayX(int isize)
    {
        this.Arr = new int[isize];
    }

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter elements of array :");

        int iCnt = 0;
        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }
    }

    public void Display()
    {
        System.out.println("Elements of the array are : ");

        int iCnt = 0;
        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            System.out.print(Arr[iCnt]+"\t");
        }
        System.out.println();
    }
}

class MyArray extends ArrayX
{
    MyArray(int size)
    {
        super(size);
    }

    public void DisplayEven()
    {
        System.out.println("Even elements of the array are : ");

        int iCnt = 0;
        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if((Arr[iCnt] % 2) == 0)
            {
                System.out.print(Arr[iCnt]+"\t");
            }
        }
        System.out.println();
    }

    public void DisplayOdd()
    {
        System.out.println("Odd elements of the array are : ");

        int iCnt = 0;
        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if((Arr[iCnt] % 2) != 0)
            {
                System.out.print(Arr[iCnt]+"\t");
            }
        }
        System.out.println();
    }
}

class program85
{
    public static void main(String arg[])
    {
    Scanner sobj = new Scanner(System.in);

    System.out.println("Enter number of elements : ");
    int size = sobj.nextInt();

    MyArray obj = new MyArray(size);
    obj.Accept();
    obj.Display();
    obj.DisplayEven();
    obj.DisplayOdd();
    }
}